folly/Checksum.cpp

```diff
@@ -29,10 +29,11 @@ namespace detail {
   #define __has_builtin(x) 0
 #endif
 
-#if (__has_builtin(__builtin_ia32_crc32qi) && \
+#if __SSE4_2__ && \
```

Comment:
As a random note, GCC does support [Function Multiversioning](https://gcc.gnu.org/onlinedocs/gcc/Function-Multiversioning.html) which would allow this check to be done at runtime rather than compile-time.